src/components/ThemeProvider/ThemeProvider.tsx

```diff
@@ -84,10 +93,21 @@ export function ThemeProvider({
     }
   }, [backgroundColor, color, isParentThemeProvider]);
 
-  const style = {...customProperties, ...(!isParentThemeProvider && {color})};
+  let style;
+
+  if (isParentThemeProvider) {
+    style = customProperties;
+  } else if (
+    !isParentThemeProvider &&
+    parentContext!.cssCustomProperties !== toString(customProperties)
+  ) {
+    style = {...customProperties, ...{color}};
```

Comment:
Rather than create an new object, perhaps we could assign to the already existing one

```suggestion
    customProperties.color = color;
    style= customProperties;
```

Reply:
This ones throws an error when modified. `Uncaught TypeError: Cannot assign to read only property 'color' of object '#<Object>'`. I believe it is because `customProperties` is the memoized value above and is now readonly. I guess we do have to create a new object here to add it.